gen/calc.py

```diff
@@ -15,6 +15,21 @@
 from pkgpanda.build import hash_checkout
 
 
+def validate_one_of(val, valid_values) -> None:
+    """Test if object `val` is a member of container `valid_values`.
+    Raise a ValueError if it is not a member. The exception message contains
+    both, the representation (__repr__) of `val` as well as the representation
+    of all items in `valid_values`.
+    """
+    if val not in valid_values:
+        options_string = ', '.join("'{}'".format(v) for v in valid_values)
```

Comment:
Practically val will always be a string, and valid_values should be some iterable list of strings. I can add python type annotations there which will assert it in the future.